hw12

irbis45 opened the pull request:
Доработана консольная команда прогрева кэша
 Добавлен кэш для популярных категорий

egerasimchuk commented on app/Http/Controllers/Admin/GrammarController.php:
```diff
@@ -39,7 +40,20 @@ public function index()
-        return view('admin.grammar_detail')->with(['grammar' => $grammar]);
+        $message="";
+        $edit=true;
+        if (Gate::allows('update-grammar', $grammar)) {
```
Будет здорово, если вы вынесите 'update-grammar' в константы, 
Gate::allows('update-grammar', $grammar) Эту логику лучше перенести во вью и воспользоваться там встроенной директивой. 
`@can(GrammarAbility::UPDATE, $grammar)`

egerasimchuk commented on app/Http/Controllers/Admin/GrammarController.php:
```diff
@@ -62,7 +76,7 @@ public function create()
     public function update(Grammar $grammar, Request $request)
     {
-        $validatedData = $request->validate([
+        $request->validate([
```
Стоит добавить проверку на права пользователя редактировать в этом методе
`$this->authorize(GrammarAbility::UPDATE, $grammar)`

egerasimchuk commented on app/Http/Controllers/Admin/GrammarController.php:
```diff
@@ -89,7 +108,7 @@ public function update(Grammar $grammar, Request $request)
     public function store(Request $request)
     {
-        $validatedData = $request->validate([
+        $request->validate([
```
Стоит добавить проверку на права пользователя создавать в этом методе
$this->authorize(GrammarAbility::STORE, $grammar)

irbis45 committed:
hw12
# Доработана консольная команда прогрева кэша
# Добавлен кэш для популярных категорий

irbis45 committed:
hw12
# добавил алиасы для понятности при подключении хэндлеров

irbis45 committed:
Merge branch 'MPshenichkin/hw11' into MPshenichkin/hw12

# Conflicts:
#	app/Console/Commands/WarmCacheNews.php
